wgengine/wgengine.go

```diff
@@ -45,31 +58,55 @@ func (rs *RouteSettings) OnlyRelevantParts() string {
 		rs.LocalAddr, rs.DNS, rs.DNSDomains, peers)
 }
 
+// Router is the TODO.
 type Router interface {
+	// Up brings the router up.
+	// TODO: more than once? after Close?
 	Up() error
-	SetRoutes(rs RouteSettings) error
+	// SetRoutes sets the routes.
+	// TODO: while running?
```

Comment:
This is called on network map updates, so periodically. It's how you get an entry in your kernel route table for a peer.